packages/core/src/components/tabs2/tabs.tsx

```diff
@@ -0,0 +1,285 @@
+     * other types of elements are ignored.
+     * @default first tab
+     */
+    defaultSelectedTabId?: TabId;
+
+    /**
+     * Unique identifier for this `Tabs` container. This will be combined with the `id` of each
+     * `Tab` child to generate ARIA accessibility attributes. IDs are required and should be
+     * unique on the page to support server-side rendering.
+     */
+    id: TabId;
+
+    /**
+     * Whether inactive tab panels should be removed from the DOM and unmounted in React.
+     * This can be a helpful performance enhancement but requires careful support for
+     * unmounting and remounting properly.
+     * @default false
+     */
+    renderActiveTabPanelOnly?: boolean;
+
+    /**
+     * Controls selected tab `id`. Providing this prop will put the component in controlled mode.
+     * Unknown ids will result in empty selection (no errors).
+     */
+    selectedTabId?: TabId;
+
+    /**
+     * Whether to show tabs stacked vertically on the left side.
+     * @default false
+     */
+    vertical?: boolean;
+
+    /**
+     * A callback function that is invoked when tabs in the tab list are clicked.
+     */
+    onChange?(newTabId: TabId, prevTabId: TabId): void;
+}
+
+export interface ITabs2State {
+    indicatorWrapperStyle?: React.CSSProperties;
+    selectedTabId?: TabId;
+}
+
+@PureRender
+export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
+    public static Tab = Tab2;
+
+    public static defaultProps: ITabs2Props = {
+        animate: true,
+        id: "",
+        renderActiveTabPanelOnly: false,
+        vertical: false,
+    };
+
+    public displayName = "Blueprint.Tabs2";
+
+    private tabElement: HTMLDivElement;
+
+    constructor(props?: ITabs2Props, context?: any) {
+        super(props, context);
+        let selectedTabId = props.selectedTabId;
+        if (selectedTabId === undefined) {
+            // select first tab in absence of user input
+            selectedTabId = props.defaultSelectedTabId === undefined
+                ? this.getTabChildren()[0].props.id
+                : props.defaultSelectedTabId;
+            // NOTE: providing an unknown ID will hide the selection
+        }
+        this.state = { selectedTabId };
+    }
+
+    public render() {
+        const { indicatorWrapperStyle, selectedTabId } = this.state;
+
+        const tabTitles = React.Children.map(this.props.children, (child) => {
+            if (isTab(child)) {
+                const { id } = child.props;
+                return (
+                    <TabTitle
+                        {...child.props}
+                        parentId={this.props.id}
+                        onClick={this.getTabClickHandler(id)}
+                        selected={id === selectedTabId}
+                    />
+                );
+            } else {
+                // TabTitle renders an <li> so let's do the same here
+                return <li>{child}</li>;
```

Comment:
sure thing. it seems we're using `li`s for the tabs but wrapping them in a `div` (not a `ul`) so might as well make them all `div`s too.